Packs/RSANetWitness_v11_1/Integrations/RSANetWitnessv115/RSANetWitnessv115_test.py

```diff
@@ -347,3 +350,182 @@ def test_paging_command(mocker):
+            self.remote_incident_id = 0
+
+    mocker.patch.object(RSANetWitnessv115, "GetRemoteDataArgs", return_value=GetRemoteDataArgsResponse())
+    mocker.patch.object(RSANetWitnessv115, "argToBoolean", return_value=True)
+    mocker.patch.object(RSANetWitnessv115, "arg_to_number", return_value=2)
+    mocker.patch.object(client, "get_incident_request", return_value={"id": 1, "status": 1, "alertCount": 2})
+    mocker.patch.object(RSANetWitnessv115, "fetch_alerts_related_incident", return_value={"alerts": [{"INC-2"}]})
+
+    res = get_remote_data_command(client, {}, {'close_incident': 0, 'import_alerts': True, 'max_alerts': 1})
+    assert res.mirrored_object == paging_data
+
+
+def test_get_modified_remote_data_command(mocker):
+    """
+        Given:
+        - client with fetch parameters
+        - args with incident attributes
+        - params
+
+        When:
+            running get_modified_remote_data_command.
+
+        Then:
+            check the updated incidents list.
+    """
+    paging_data = ['INC-1']
+
+    class GetModifiedRemoteDataArgsResponse:
+        def __init__(self) -> dict:
+            self.last_update = "1694188115"
+
+    mocker.patch.object(RSANetWitnessv115, "GetModifiedRemoteDataArgs", return_value=GetModifiedRemoteDataArgsResponse())
+    mocker.patch.object(client, "get_incident_request", return_value={"id": "INC-1", "status": 1, "alertCount": 1})
+    mocker.patch.object(RSANetWitnessv115, "paging_command", return_value=({}, [{"id": "INC-1", "lastUpdated": 1694188116}]))
+    mocker.patch.object(RSANetWitnessv115, "clean_old_inc_context", return_value=False)
+    mocker.patch.object(RSANetWitnessv115, "get_integration_context", return_value={"INC-1": {"alertCount": 2, "eventCount": 2}})
+
+    res = get_modified_remote_data_command(client, {}, {"max_fetch": 2, "max_alerts": 2, "max_mirror_time": 0})
+    assert res.modified_incident_ids == paging_data
+
+
+def test_struct_inc_context():
+    """
+        Given:
+        - Alert count
+        - Event Count
+        - Created boolean
+
+        When:
+            running get_remote_data_command.
+
+        Then:
+            Create base incident structure.
+    """
+    expected_result = {"alertCount": 1, "eventCount": 1, "Created": True}
+    assert struct_inc_context(1, 1, True) == expected_result
+
+
+def test_clean_old_inc_context(mocker):
+    """
+        Given:
+        - Max mirror time aggregration (int for days)
+
+        When:
+            running get_modified_remote_data_command.
+
+        Then:
+            Clean old incident.
+    """
+    from datetime import datetime, timedelta
+    max_time_mirror_inc = 24
+
+    mocker.patch.object(demisto, "getIntegrationContext", return_value={"IncidentsDataCount": {}})
+    mocker.patch.object(RSANetWitnessv115, "arg_to_datetime",
+                        return_value=datetime.now() + timedelta(days=max_time_mirror_inc + 1))
+
+    assert clean_old_inc_context(max_time_mirror_inc) is None
```

Comment:
Please mock the getIntegrationContextand make sure that the needed incident was deleted from the context.

Reply:
I created others functions to add some tests:
- test_get_remote_data_command_with_closed_xsoar_incident
- test_get_remote_data_command_with_new_xsoar_incident_entries

Reply:
You mean setIntegrationContext ?